File: codigocomcafe-back/src/main/java/br/com/codigocomcafe/model/PostModel.java
package br.com.codigocomcafe.model;

import br.com.codigocomcafe.enumerador.Status;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "tb_post")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PostModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String titulo;

    private String autor;

    @Column(columnDefinition = "TEXT")
    @NotBlank
    private String conteudo;

    private String imagem; // nome do arquivo
    private String video; // nome do arquivo

    @Enumerated(EnumType.STRING)
    private Status status;

    private LocalDateTime dataCriacao;
    private LocalDateTime dataAtualizacao;
    private LocalDateTime dataPublicacao;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id", nullable = false)
    private CategoriaModel categoriaModel;

    @PrePersist
    protected void onCreate() {
        dataCriacao = LocalDateTime.now();
        dataAtualizacao = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        dataAtualizacao = LocalDateTime.now();
    }
}

File: codigocomcafe-front/src/app/model/PostModel.ts
import { CategoriaModel } from "./CategoriaModel";

// Enum para status
export type Status = 'RASCUNHO' | 'PUBLICADO' | 'EXCLUIDO';

export class PostModel {
    id?: number;
    titulo?: string;
    autor?: string;
    conteudo?: string;
    imagem?: File | undefined; // agora é File, compatível com upload
    video?: File | undefined;  // agora é File, compatível com upload
    status?: Status;
    dataCriacao?: Date;    // tipo Date em vez de string
    dataAtualizacao?: Date;
    dataPublicacao?: Date;
    categoria?: CategoriaModel;
}
